JExample/trunk/src/main/ch/unibe/jexample/internal/ExampleGraph.java

```diff
@@ -102,11 +102,22 @@ public Example findExample(MethodReference ref) {
     public void run(ExampleClass testClass, RunNotifier notifier) {
         if (!anyHasBeenRun) onStartRunning();
         anyHasBeenRun = true;
+                
+        for(Method m : testClass.getImplementingClass().getMethods()){
+            if (!m.isAnnotationPresent(BeforeTestRun.class)) continue;
+            Reflection.invoke(m, null);
+        }
+        
         for (Example e: this.getExamples()) {
             if (testClass.contains(e)) {
                 e.run(notifier);
             }
         }
+        
+        for(Method m : testClass.getImplementingClass().getMethods()){

```

Comment:
As far as I remember other example classes may depend on cached results from `testClass` even after these `@ AfterTestRun` functions have been run, are you okay with this limitation? If yes we might even consider running the `@AfterClass` annotations here.